core/filter_api.php

```diff
@@ -992,25 +992,29 @@ function filter_deserialize( $p_serialized_filter ) {
 	# in this case, the filter version field inside the array is to be used
 	# and if not present, set the current filter version
 
-	# check to see if new cookie is needed
+	# check fitler version mark
 	$t_setting_arr = explode( '#', $p_serialized_filter, 2 );
-	if( ( $t_setting_arr[0] == 'v1' ) || ( $t_setting_arr[0] == 'v2' ) || ( $t_setting_arr[0] == 'v3' ) || ( $t_setting_arr[0] == 'v4' ) ) {
+	$t_version_string = $t_setting_arr[0];
+	if( in_array( $t_version_string, array( 'v1', 'v2', 'v3', 'v4' ) ) ) {
```

Comment:
Should the upgrade logic here be part of the filter_upgrade method that you added?

Reply:
`filter_version_upgrade()` deals with the contents of the filter array.
That code in `filter_deserialize()` deals with how the data is physically stored in db. It's not actually upgrading anything, only trying to read it properly

Reply:
Makes sense.